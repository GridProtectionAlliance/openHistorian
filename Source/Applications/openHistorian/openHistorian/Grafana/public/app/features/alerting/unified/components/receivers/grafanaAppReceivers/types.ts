import { Receiver } from '../../../../../../plugins/datasource/alertmanager/types';
import { SupportedPlugin } from '../../../types/pluginBridges';

export interface AmRouteReceiver {
  label: string;
  value: string;
  grafanaAppReceiverType?: SupportedPlugin;
}

export interface ReceiverWithTypes extends Receiver {
  grafanaAppReceiverType?: SupportedPlugin;
}
export const GRAFANA_APP_RECEIVERS_SOURCE_IMAGE: Record<SupportedPlugin, string> = {
  [SupportedPlugin.OnCall]: 'public/img/alerting/oncall_logo.svg',

  [SupportedPlugin.Incident]: '',
  [SupportedPlugin.MachineLearning]: '',
};
