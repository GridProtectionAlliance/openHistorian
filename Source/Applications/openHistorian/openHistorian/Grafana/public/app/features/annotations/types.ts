import { AnnotationEvent, PanelData, TimeRange } from '@grafana/data';

import { DashboardModel, PanelModel } from '../dashboard/state';

export interface AnnotationQueryOptions {
  dashboard: DashboardModel;
  panel: PanelModel;
  range: TimeRange;
}

export interface AnnotationQueryResponse {
  /**
   * The processed annotation events
   */
  events?: AnnotationEvent[];

  /**
   * The original panel response
   */
  panelData?: PanelData;
}

export interface AnnotationTag {
  /**
   * The tag name
   */
  tag: string;
  /**
   * The number of occurrences of that tag
   */
  count: number;
}

export interface AnnotationTagsResponse {
  result: {
    tags: AnnotationTag[];
  };
}
