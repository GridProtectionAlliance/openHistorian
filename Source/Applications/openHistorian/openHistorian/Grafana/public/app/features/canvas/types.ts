import { ColorDimensionConfig, ResourceDimensionConfig } from 'app/features/dimensions/types';

export interface Placement {
  top?: number;
  left?: number;
  right?: number;
  bottom?: number;

  width?: number;
  height?: number;
}

export interface Constraint {
  horizontal?: HorizontalConstraint;
  vertical?: VerticalConstraint;
}

export enum HorizontalConstraint {
  Left = 'left',
  Right = 'right',
  LeftRight = 'leftright',
  Center = 'center',
  Scale = 'scale',
}

export enum VerticalConstraint {
  Top = 'top',
  Bottom = 'bottom',
  TopBottom = 'topbottom',
  Center = 'center',
  Scale = 'scale',
}

export enum BackgroundImageSize {
  Original = 'original',
  Contain = 'contain',
  Cover = 'cover',
  Fill = 'fill',
  Tile = 'tile',
}

export interface BackgroundConfig {
  color?: ColorDimensionConfig;
  image?: ResourceDimensionConfig;
  size?: BackgroundImageSize;
}

export interface LineConfig {
  color?: ColorDimensionConfig;
  width?: number;
}

export enum QuickPlacement {
  Top = 'top',
  Bottom = 'bottom',
  Left = 'left',
  Right = 'right',
  HorizontalCenter = 'hcenter',
  VerticalCenter = 'vcenter',
}
