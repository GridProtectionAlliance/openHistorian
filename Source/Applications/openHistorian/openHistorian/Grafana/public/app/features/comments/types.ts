export interface MessagePacket {
  event: string;
  commentCreated: Message;
}

export interface Message {
  id: number;
  content: string;
  created: number;
  userId: number;
  user: User;
}

// TODO: Interface may exist elsewhere
export interface User {
  id: number;
  name: string;
  login: string;
  email: string;
  avatarUrl: string;
}
