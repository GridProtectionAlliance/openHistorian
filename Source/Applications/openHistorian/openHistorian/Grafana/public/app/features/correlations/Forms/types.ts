import { CorrelationConfig } from '../types';

export interface FormDTO {
  sourceUID: string;
  targetUID: string;
  label: string;
  description: string;
  config: CorrelationConfig;
}

export type EditFormDTO = Omit<FormDTO, 'targetUID' | 'sourceUID'>;
