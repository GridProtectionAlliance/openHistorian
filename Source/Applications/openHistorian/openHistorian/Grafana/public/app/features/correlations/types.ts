import { DataLinkTransformationConfig } from '@grafana/data';

export interface AddCorrelationResponse {
  correlation: Correlation;
}

export type GetCorrelationsResponse = Correlation[];

type CorrelationConfigType = 'query';

export interface CorrelationConfig {
  field: string;
  target: object;
  type: CorrelationConfigType;
  transformations?: DataLinkTransformationConfig[];
}

export interface Correlation {
  uid: string;
  sourceUID: string;
  targetUID: string;
  label?: string;
  description?: string;
  config: CorrelationConfig;
}

export type RemoveCorrelationParams = Pick<Correlation, 'sourceUID' | 'uid'>;
export type CreateCorrelationParams = Omit<Correlation, 'uid'>;
export type UpdateCorrelationParams = Omit<Correlation, 'targetUID'>;
