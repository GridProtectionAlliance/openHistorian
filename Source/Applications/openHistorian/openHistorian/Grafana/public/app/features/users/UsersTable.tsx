import React, { FC } from 'react';
import { OrgUser } from 'app/types';

export interface Props {
  users: OrgUser[];
  onRoleChange: (role: string, user: OrgUser) => void;
  onRemoveUser: (user: OrgUser) => void;
}

const UsersTable: FC<Props> = props => {
  const { users, onRoleChange, onRemoveUser } = props;

  return (
    <table className="filter-table form-inline">
      <thead>
        <tr>
          <th />
          <th>Login</th>
          <th>Email</th>
          <th>Name</th>
          <th>Seen</th>
          <th>Role</th>
          <th style={{ width: '34px' }} />
        </tr>
      </thead>
      <tbody>
        {users.map((user, index) => {
          return (
            <tr key={`${user.userId}-${index}`}>
              <td className="width-4 text-center">
                <img className="filter-table__avatar" src={user.avatarUrl} />
              </td>
              <td>{user.login}</td>
              <td>
                <span className="ellipsis">{user.email}</span>
              </td>
              <td>{user.name}</td>
              <td>{user.lastSeenAtAge}</td>
              <td>
                <div className="gf-form-select-wrapper width-12">
                  <select
                    value={user.role}
                    className="gf-form-input"
                    onChange={event => onRoleChange(event.target.value, user)}
                  >
                    {['Viewer', 'Editor', 'Admin'].map((option, index) => {
                      return (
                        <option value={option} key={`${option}-${index}`}>
                          {option}
                        </option>
                      );
                    })}
                  </select>
                </div>
              </td>
              <td>
                <div onClick={() => onRemoveUser(user)} className="btn btn-danger btn-small">
                  <i className="fa fa-remove" />
                </div>
              </td>
            </tr>
          );
        })}
      </tbody>
    </table>
  );
};

export default UsersTable;
