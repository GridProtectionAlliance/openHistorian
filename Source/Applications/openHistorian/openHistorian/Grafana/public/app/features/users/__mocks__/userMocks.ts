import { OrgRole, OrgUser } from 'app/types';

export const getMockUsers = (amount: number) => {
  const users = [];

  for (let i = 0; i <= amount; i++) {
    users.push({
      avatarUrl: 'url/to/avatar',
      email: `user-${i}@test.com`,
      name: `user-${i} test`,
      lastSeenAt: '2018-10-01',
      lastSeenAtAge: '',
      login: `user-${i}`,
      orgId: 1,
      role: 'Admin',
      userId: i,
    });
  }

  return users as OrgUser[];
};

export const getMockUser = () => {
  return {
    avatarUrl: 'url/to/avatar',
    email: `user@test.com`,
    name: 'user test',
    lastSeenAt: '2018-10-01',
    lastSeenAtAge: '',
    login: `user`,
    orgId: 1,
    role: 'Admin' as OrgRole,
    userId: 2,
  } as OrgUser;
};

export const getMockInvitees = (amount: number) => {
  const invitees = [];

  for (let i = 0; i <= amount; i++) {
    invitees.push({
      code: `asdfasdfsadf-${i}`,
      createdOn: '2018-10-02',
      email: `invitee-${i}@test.com`,
      emailSent: true,
      emailSentOn: '2018-10-02',
      id: i,
      invitedByEmail: 'admin@grafana.com',
      invitedByLogin: 'admin',
      invitedByName: 'admin',
      name: `invitee-${i}`,
      orgId: 1,
      role: 'viewer',
      status: 'not accepted',
      url: `localhost/invite/${i}`,
    });
  }

  return invitees;
};
