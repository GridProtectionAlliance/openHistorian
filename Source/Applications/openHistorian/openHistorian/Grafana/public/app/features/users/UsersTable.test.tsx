import React from 'react';
import { shallow } from 'enzyme';
import UsersTable, { Props } from './UsersTable';
import { OrgUser } from 'app/types';
import { getMockUsers } from './__mocks__/userMocks';
import { ConfirmModal } from '@grafana/ui';

const setup = (propOverrides?: object) => {
  const props: Props = {
    users: [] as OrgUser[],
    onRoleChange: jest.fn(),
    onRemoveUser: jest.fn(),
  };

  Object.assign(props, propOverrides);

  return shallow(<UsersTable {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render users table', () => {
    const wrapper = setup({
      users: getMockUsers(5),
    });

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Remove modal', () => {
  it('should render correct amount', () => {
    const wrapper = setup({
      users: getMockUsers(3),
    });
    expect(wrapper.find(ConfirmModal).length).toEqual(4);
  });
});
