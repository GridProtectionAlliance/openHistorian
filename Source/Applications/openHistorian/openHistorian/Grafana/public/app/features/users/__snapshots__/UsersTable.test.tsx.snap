// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<table
  className="filter-table form-inline"
>
  <thead>
    <tr>
      <th />
      <th>
        Login
      </th>
      <th>
        Email
      </th>
      <th>
        Name
      </th>
      <th>
        Seen
      </th>
      <th>
        Role
      </th>
      <th
        style={
          Object {
            "width": "34px",
          }
        }
      />
    </tr>
  </thead>
  <tbody />
</table>
`;

exports[`Render should render users table 1`] = `
<table
  className="filter-table form-inline"
>
  <thead>
    <tr>
      <th />
      <th>
        Login
      </th>
      <th>
        Email
      </th>
      <th>
        Name
      </th>
      <th>
        Seen
      </th>
      <th>
        Role
      </th>
      <th
        style={
          Object {
            "width": "34px",
          }
        }
      />
    </tr>
  </thead>
  <tbody>
    <tr
      key="0-0"
    >
      <td
        className="width-4 text-center"
      >
        <img
          className="filter-table__avatar"
          src="url/to/avatar"
        />
      </td>
      <td>
        user-0
      </td>
      <td>
        <span
          className="ellipsis"
        >
          user-0@test.com
        </span>
      </td>
      <td>
        user-0 test
      </td>
      <td />
      <td>
        <div
          className="gf-form-select-wrapper width-12"
        >
          <select
            className="gf-form-input"
            onChange={[Function]}
            value="Admin"
          >
            <option
              key="Viewer-0"
              value="Viewer"
            >
              Viewer
            </option>
            <option
              key="Editor-1"
              value="Editor"
            >
              Editor
            </option>
            <option
              key="Admin-2"
              value="Admin"
            >
              Admin
            </option>
          </select>
        </div>
      </td>
      <td>
        <div
          className="btn btn-danger btn-small"
          onClick={[Function]}
        >
          <i
            className="fa fa-remove"
          />
        </div>
      </td>
    </tr>
    <tr
      key="1-1"
    >
      <td
        className="width-4 text-center"
      >
        <img
          className="filter-table__avatar"
          src="url/to/avatar"
        />
      </td>
      <td>
        user-1
      </td>
      <td>
        <span
          className="ellipsis"
        >
          user-1@test.com
        </span>
      </td>
      <td>
        user-1 test
      </td>
      <td />
      <td>
        <div
          className="gf-form-select-wrapper width-12"
        >
          <select
            className="gf-form-input"
            onChange={[Function]}
            value="Admin"
          >
            <option
              key="Viewer-0"
              value="Viewer"
            >
              Viewer
            </option>
            <option
              key="Editor-1"
              value="Editor"
            >
              Editor
            </option>
            <option
              key="Admin-2"
              value="Admin"
            >
              Admin
            </option>
          </select>
        </div>
      </td>
      <td>
        <div
          className="btn btn-danger btn-small"
          onClick={[Function]}
        >
          <i
            className="fa fa-remove"
          />
        </div>
      </td>
    </tr>
    <tr
      key="2-2"
    >
      <td
        className="width-4 text-center"
      >
        <img
          className="filter-table__avatar"
          src="url/to/avatar"
        />
      </td>
      <td>
        user-2
      </td>
      <td>
        <span
          className="ellipsis"
        >
          user-2@test.com
        </span>
      </td>
      <td>
        user-2 test
      </td>
      <td />
      <td>
        <div
          className="gf-form-select-wrapper width-12"
        >
          <select
            className="gf-form-input"
            onChange={[Function]}
            value="Admin"
          >
            <option
              key="Viewer-0"
              value="Viewer"
            >
              Viewer
            </option>
            <option
              key="Editor-1"
              value="Editor"
            >
              Editor
            </option>
            <option
              key="Admin-2"
              value="Admin"
            >
              Admin
            </option>
          </select>
        </div>
      </td>
      <td>
        <div
          className="btn btn-danger btn-small"
          onClick={[Function]}
        >
          <i
            className="fa fa-remove"
          />
        </div>
      </td>
    </tr>
    <tr
      key="3-3"
    >
      <td
        className="width-4 text-center"
      >
        <img
          className="filter-table__avatar"
          src="url/to/avatar"
        />
      </td>
      <td>
        user-3
      </td>
      <td>
        <span
          className="ellipsis"
        >
          user-3@test.com
        </span>
      </td>
      <td>
        user-3 test
      </td>
      <td />
      <td>
        <div
          className="gf-form-select-wrapper width-12"
        >
          <select
            className="gf-form-input"
            onChange={[Function]}
            value="Admin"
          >
            <option
              key="Viewer-0"
              value="Viewer"
            >
              Viewer
            </option>
            <option
              key="Editor-1"
              value="Editor"
            >
              Editor
            </option>
            <option
              key="Admin-2"
              value="Admin"
            >
              Admin
            </option>
          </select>
        </div>
      </td>
      <td>
        <div
          className="btn btn-danger btn-small"
          onClick={[Function]}
        >
          <i
            className="fa fa-remove"
          />
        </div>
      </td>
    </tr>
    <tr
      key="4-4"
    >
      <td
        className="width-4 text-center"
      >
        <img
          className="filter-table__avatar"
          src="url/to/avatar"
        />
      </td>
      <td>
        user-4
      </td>
      <td>
        <span
          className="ellipsis"
        >
          user-4@test.com
        </span>
      </td>
      <td>
        user-4 test
      </td>
      <td />
      <td>
        <div
          className="gf-form-select-wrapper width-12"
        >
          <select
            className="gf-form-input"
            onChange={[Function]}
            value="Admin"
          >
            <option
              key="Viewer-0"
              value="Viewer"
            >
              Viewer
            </option>
            <option
              key="Editor-1"
              value="Editor"
            >
              Editor
            </option>
            <option
              key="Admin-2"
              value="Admin"
            >
              Admin
            </option>
          </select>
        </div>
      </td>
      <td>
        <div
          className="btn btn-danger btn-small"
          onClick={[Function]}
        >
          <i
            className="fa fa-remove"
          />
        </div>
      </td>
    </tr>
    <tr
      key="5-5"
    >
      <td
        className="width-4 text-center"
      >
        <img
          className="filter-table__avatar"
          src="url/to/avatar"
        />
      </td>
      <td>
        user-5
      </td>
      <td>
        <span
          className="ellipsis"
        >
          user-5@test.com
        </span>
      </td>
      <td>
        user-5 test
      </td>
      <td />
      <td>
        <div
          className="gf-form-select-wrapper width-12"
        >
          <select
            className="gf-form-input"
            onChange={[Function]}
            value="Admin"
          >
            <option
              key="Viewer-0"
              value="Viewer"
            >
              Viewer
            </option>
            <option
              key="Editor-1"
              value="Editor"
            >
              Editor
            </option>
            <option
              key="Admin-2"
              value="Admin"
            >
              Admin
            </option>
          </select>
        </div>
      </td>
      <td>
        <div
          className="btn btn-danger btn-small"
          onClick={[Function]}
        >
          <i
            className="fa fa-remove"
          />
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
